jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -469,66 +466,130 @@ protected void sendWelcome(HttpContent content, String pathInContext, boolean en
             sendDirectory(request, response, content, callback, pathInContext);
     }
 
-    public enum WelcomeActionType
+    /**
+     * <p>How welcome targets should be processed.</p>
+     */
+    public enum WelcomeMode
     {
+        /**
+         * The welcome target is used in an HTTP redirect
+         * as the value of the {@code Location} header.
+         */
         REDIRECT,
-        SERVE
+        /**
+         * The welcome target is served either as a real file
+         * or forwarded to be processed again.
```

Comment:
what does "or forwarded to be processed again" mean?

Reply:
`ServletResourceService` overrides `serveWelcome()` to forward the welcome file, so it is re-processed as it goes again through the filters, etc.
I'm fine improving the text, you have suggestions?